pkg/sql/sem/builtins/builtins.go

```diff
@@ -6384,6 +6384,9 @@ table's zone configuration this will return NULL.`,
 						"can only deserialize matching session users",
 					)
 				}
+				if err := evalCtx.Planner.CheckCanBecomeUser(evalCtx.Context, sd.User()); err != nil {
+					return nil, errors.Wrap(err, "cannot SET ROLE to user")
```

Comment:
might be fine to return this error without extra wrapping. (the SET ROLE might be misleading anyway)